src/Core/Framework/App/Payload/AppPayloadServiceHelper.php

```diff
@@ -67,6 +68,8 @@ public function encode(SourcedPayloadInterface $payload): array
                 }
 
                 $array[$propertyName] = $salesChannelContext;
+            } elseif ($property instanceof RequestDataBag) {
+                $array[$propertyName] = $property->all();
```

Comment:
Would add a `continue` here. Just to be sure, nothing else will be done
```suggestion
                $array[$propertyName] = $property->all();
                
                continue;
```

Reply:
I don't think this is necessary; the original if-branch doesn't have one either

Reply:
Yes, it is not necessary. But in my opinion, the `continue` is missing in the `if` above as well.
These are completed tasks. Everyone knows, on the first look, that all things are done. And there will be no side effects on further and/or changes to the code that may occur in the future.

----

Furthermore, I would have liked to move the "encode SalesChannelContext" logic into a separate function (What does not have to be a task for this PR). It would reduce the visible complexity and depth of the `foreach`.

Something like that
```php
if ($property instanceof SalesChannelContext) {
    $array[$propertyName] = $this->encodeSalesChannelContext($property);

    continue;
} elseif ($property instanceof RequestDataBag) {
    $array[$propertyName] = $property->all();

    continue;
}

if (!$property instanceof Entity) {
    continue;
}

$array[$propertyName] = $this->encodeEntity($property);
```
or even
```php
if ($property instanceof SalesChannelContext) {
    $array[$propertyName] = $this->encodeSalesChannelContext($property);

} elseif ($property instanceof RequestDataBag) {
    $array[$propertyName] = $property->all();

} elseif ($property instanceof Entity) {
    $array[$propertyName] = $this->encodeEntity($property);
}


```

Reply:
@mkreusch I would really appreciate it if you could implement my suggestion for better understanding and readability.

This seems to be the best one
```php
if ($property instanceof SalesChannelContext) {
    $array[$propertyName] = $this->encodeSalesChannelContext($property);

} elseif ($property instanceof RequestDataBag) {
    $array[$propertyName] = $property->all();

} elseif ($property instanceof Entity) {
    $array[$propertyName] = $this->encodeEntity($property);
}
```

Reply:
@En0Ma1259 Sure, I'll update that 👍 I was not too much of a fan of the original structure either, but in this PR just tried to stick to what was already there.

Reply:
It's now updated and checked in our local project that this implementation passes all parameters to the app server.

<img width="1372" height="1154" alt="image" src="https://github.com/user-attachments/assets/fafa8d80-d9fc-4198-8d75-56454ed85913" />